File: Level-2/5.java
class Solution {
    public int[] d;
    public int solution(int n) {
        d = new int[n + 1];
        int answer = fibo(n);
        return answer;
    }
    
    public int fibo(int n){
        if(n == 1 || n == 2) {
            return n;
        }
        if(d[n] != 0){
            return d[n] % 1000000007;
        }
        d[n] = fibo(n - 1) + fibo(n - 2);
        return d[n] % 1000000007;
    }
}

File: Level-2/5.js
function solution(board) {
    let answer = 0;
    
    const row = board.length;
    const col = board[0].length;
    
    if(row === 1 || col === 1) return board[0][0];
    
    for(let i = 1; i < row; i++) {
        for(let j = 1; j < col; j++) {
            if(board[i][j] === 1) {
                board[i][j] = Math.min(board[i-1][j], board[i][j-1], board[i-1][j-1]) + 1;
                answer = Math.max(answer, board[i][j]);
            }
        }
    }
    
    return answer * answer;
}
